src/validator.cpp

```diff
@@ -1035,20 +1057,26 @@ void Validator::ValidatorImpl::validateVariableInterface(const VariablePtr &vari
     }
 }
 
-void Validator::ValidatorImpl::validateEquivalenceUnits(const ModelPtr &model, const VariablePtr &variable)
+void Validator::ValidatorImpl::validateEquivalenceUnits(const ModelPtr &model, const VariablePtr &variable, VariableMap &alreadyReported)
 {
     std::string hints;
     for (size_t index = 0; index < variable->equivalentVariableCount(); ++index) {
         auto equivalentVariable = variable->equivalentVariable(index);
         double multiplier = 0.0;
         if (!unitsAreEquivalent(model, variable, equivalentVariable, hints, multiplier)) {
-            auto unitsName = variable->units() == nullptr ? "" : variable->units()->name();
-            auto equivalentUnitsName = equivalentVariable->units() == nullptr ? "" : equivalentVariable->units()->name();
-            ErrorPtr err = Error::create();
-            err->setDescription("Variable '" + variable->name() + "' has units of '" + unitsName + "' and an equivalent variable '" + equivalentVariable->name() + "' with non-matching units of '" + equivalentUnitsName + "'. The mismatch is: " + hints);
-            err->setModel(model);
-            err->setKind(Error::Kind::UNITS);
-            mValidator->addError(err);
+            VariablePair reversePair = std::make_pair(equivalentVariable, variable);
+            auto it = std::find(alreadyReported.begin(), alreadyReported.end(), reversePair);
+            if (it == alreadyReported.end()) {
+                VariablePair pair = std::make_pair(variable, equivalentVariable);
+                alreadyReported.push_back(pair);
+                auto unitsName = variable->units() == nullptr ? "" : variable->units()->name();
+                auto equivalentUnitsName = equivalentVariable->units() == nullptr ? "" : equivalentVariable->units()->name();
+                ErrorPtr err = Error::create();
+                err->setDescription("Variable '" + variable->name() + "' has units of '" + unitsName + "' and an equivalent variable '" + equivalentVariable->name() + "' with non-matching units of '" + equivalentUnitsName + "'. The mismatch is: " + hints);
+                err->setModel(model);
+                err->setKind(Error::Kind::UNITS);
```

Comment:
Do we need a better error code that just UNITS here?  The stuff under that spec heading has no relevance to this situation ...?

Reply:
This one comes under "interpretation of map_variables" I think.